File: kirun-js/__tests__/engine/function/system/array/MaxTest.ts
import { Max } from '../../../../../src/engine/function/system/array/Max';
import { FunctionExecutionParameters } from '../../../../../src/engine/runtime/FunctionExecutionParameters';

let max: Max = new Max();

test('max test 1 ', async () => {
    let arr: any[] = [];
    arr.push(null);
    arr.push(12);

    let fep: FunctionExecutionParameters = new FunctionExecutionParameters().setArguments(
        new Map([[Max.PARAMETER_ARRAY_SOURCE_PRIMITIVE.getParameterName(), arr]]),
    );

    expect(
        (await max.execute(fep)).allResults()[0].getResult().get(Max.EVENT_RESULT_ANY.getName()),
    ).toBe(12);
});

test('max test 2 ', async () => {
    let arr: any[] = [];

    let fep: FunctionExecutionParameters = new FunctionExecutionParameters().setArguments(
        new Map([[Max.PARAMETER_ARRAY_SOURCE_PRIMITIVE.getParameterName(), arr]]),
    );

    await expect(max.execute(fep)).rejects.toThrow();
});

test('max test 3', async () => {
    let arr: any[] = [];
    arr.push(12);
    arr.push(15);
    arr.push(null);
    arr.push(98);
    arr.push(1);

    let fep: FunctionExecutionParameters = new FunctionExecutionParameters().setArguments(
        new Map([[Max.PARAMETER_ARRAY_SOURCE_PRIMITIVE.getParameterName(), arr]]),
    );
    expect((await max.execute(fep)).allResults()[0].getResult().get('output')).toBe(98);
});

test('Max test 4', async () => {
    let arr: any[] = [];
    arr.push('nocode');
    arr.push('NoCode');
    arr.push('platform');
    let fep: FunctionExecutionParameters = new FunctionExecutionParameters().setArguments(
        new Map([[Max.PARAMETER_ARRAY_SOURCE_PRIMITIVE.getParameterName(), arr]]),
    );
    expect((await max.execute(fep)).allResults()[0].getResult().get('output')).toBe('platform');
});

test('Max test 6', async () => {
    let fep: FunctionExecutionParameters = new FunctionExecutionParameters().setArguments(
        new Map([[Max.PARAMETER_ARRAY_SOURCE_PRIMITIVE.getParameterName(), null]]),
    );
    await expect(max.execute(fep)).rejects.toThrow();
});

test('Max test 5', async () => {
    let arr: any[] = [];

    arr.push(456);
    arr.push('nocode');

    arr.push('NoCode');
    arr.push('platform');
    arr.push(123);

    let fep: FunctionExecutionParameters = new FunctionExecutionParameters().setArguments(
        new Map([[Max.PARAMETER_ARRAY_SOURCE_PRIMITIVE.getParameterName(), arr]]),
    );
    expect((await max.execute(fep)).allResults()[0].getResult().get('output')).toBe('platform');
});

test('Max test 7', async () => {
    let arr1: any[] = [];
    arr1.push('c');
    arr1.push('r');
    arr1.push('d');
    arr1.push('s');
    let fep: FunctionExecutionParameters = new FunctionExecutionParameters().setArguments(
        new Map([[Max.PARAMETER_ARRAY_SOURCE_PRIMITIVE.getParameterName(), arr1]]),
    );

    expect((await max.execute(fep)).allResults()[0].getResult().get('output')).toBe('s');
});

test('Max test 8', async () => {
    let arr: any[] = ['surendhar'];
    let fep: FunctionExecutionParameters = new FunctionExecutionParameters().setArguments(
        new Map([[Max.PARAMETER_ARRAY_SOURCE_PRIMITIVE.getParameterName(), arr]]),
    );

    expect((await max.execute(fep)).allResults()[0].getResult().get('output')).toBe('surendhar');
});

File: kirun-java/src/test/java/com/fincity/nocode/kirun/engine/function/system/array/MaxTest.java
package com.fincity.nocode.kirun.engine.function.system.array;

import java.util.Map;

import org.junit.jupiter.api.Test;

import com.fincity.nocode.kirun.engine.exception.KIRuntimeException;
import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveFunctionRepository;
import com.fincity.nocode.kirun.engine.repository.reactive.KIRunReactiveSchemaRepository;
import com.fincity.nocode.kirun.engine.runtime.reactive.ReactiveFunctionExecutionParameters;
import com.google.gson.JsonArray;
import com.google.gson.JsonNull;
import com.google.gson.JsonPrimitive;

import reactor.test.StepVerifier;

class MaxTest {

	@Test
	void test() {
		var arr = new JsonArray();
		arr.add(12);

		Max max = new Max();

		ReactiveFunctionExecutionParameters fep = new ReactiveFunctionExecutionParameters(
				new KIRunReactiveFunctionRepository(), new KIRunReactiveSchemaRepository())
				.setArguments(Map.of("source", arr));

		StepVerifier.create(max.execute(fep))
				.expectNextMatches(r -> r.next().getResult().get("output").equals(new JsonPrimitive(12)))
				.verifyComplete();

		var arr1 = new JsonArray();
		ReactiveFunctionExecutionParameters fep1 = new ReactiveFunctionExecutionParameters(
				new KIRunReactiveFunctionRepository(), new KIRunReactiveSchemaRepository())
				.setArguments(Map.of("source", arr1));

		StepVerifier.create(max.execute(fep1))
				.verifyError(KIRuntimeException.class);
	}

	@Test
	void test2() {
		var arr = new JsonArray();
		arr.add(12);
		arr.add(15);
		arr.add(98);
		arr.add(1);

		Max max = new Max();

		ReactiveFunctionExecutionParameters fep = new ReactiveFunctionExecutionParameters(
				new KIRunReactiveFunctionRepository(), new KIRunReactiveSchemaRepository())
				.setArguments(Map.of("source", arr));

		StepVerifier.create(max.execute(fep))
				.expectNextMatches(r -> r.next().getResult().get("output").equals(new JsonPrimitive(98)))
				.verifyComplete();

		var arr1 = new JsonArray();

		arr1.add('c');
		arr1.add('r');
		arr1.add('d');
		arr1.add('s');
		ReactiveFunctionExecutionParameters fep1 = new ReactiveFunctionExecutionParameters(
				new KIRunReactiveFunctionRepository(), new KIRunReactiveSchemaRepository())
				.setArguments(Map.of("source", arr1));

		StepVerifier.create(max.execute(fep1))
				.expectNextMatches(r -> r.next().getResult().get("output").equals(new JsonPrimitive('s')))
				.verifyComplete();

	}

	@Test
	void test3() {
		var arr = new JsonArray();
		Max max = new Max();

		ReactiveFunctionExecutionParameters fep = new ReactiveFunctionExecutionParameters(
				new KIRunReactiveFunctionRepository(), new KIRunReactiveSchemaRepository())
				.setArguments(Map.of("source", arr));

		StepVerifier.create(max.execute(fep))
				.verifyError(KIRuntimeException.class);
	}

	@Test
	void test4() {
		var arr = new JsonArray();
		arr.add("nocode");
		arr.add("NoCode");
		arr.add("platform");

		Max max = new Max();

		ReactiveFunctionExecutionParameters fep = new ReactiveFunctionExecutionParameters(
				new KIRunReactiveFunctionRepository(), new KIRunReactiveSchemaRepository())
				.setArguments(Map.of("source", arr));

		StepVerifier.create(max.execute(fep))
				.expectNextMatches(r -> r.next().getResult().get("output").equals(new JsonPrimitive("platform")))
				.verifyComplete();
	}

	@Test
	void test5() {
		var arr = new JsonArray();
		arr.add(456);
		arr.add("nocode");
		arr.add("NoCode");
		arr.add("platform");
		arr.add(123);

		Max max = new Max();

		ReactiveFunctionExecutionParameters fep = new ReactiveFunctionExecutionParameters(
				new KIRunReactiveFunctionRepository(), new KIRunReactiveSchemaRepository())
				.setArguments(Map.of("source", arr));

		StepVerifier.create(max.execute(fep))
				.expectNextMatches(r -> r.next().getResult().get("output").equals(new JsonPrimitive("platform")))
				.verifyComplete();
	}

	@Test
	void test6() {
		Max max = new Max();

		ReactiveFunctionExecutionParameters fep = new ReactiveFunctionExecutionParameters(
				new KIRunReactiveFunctionRepository(), new KIRunReactiveSchemaRepository())
				.setArguments(Map.of("source", JsonNull.INSTANCE));

		StepVerifier.create(max.execute(fep))
				.verifyError(KIRuntimeException.class);
	}
}
